src/api.ts

```
@@ -1,10 +1,11 @@
 import { MovieData } from './types/MovieData';
 import { ResponseError } from './types/ReponseError';
 
-const API_URL = 'https://www.omdbapi.com/?apikey=your-key';
+const API_KEY = import.meta.env.VITE_OMDB_API_KEY;
```

Potential runtime issue if the API key is missing. Reading the key directly from import.meta.env is fine, but when VITE_OMDB_API_KEY is undefined the API_URL will contain the string "undefined" and requests will silently fail. Consider validating the variable and returning/throwing a clear error early, for example:

```ts
if (!API_KEY) {
  throw new Error('VITE_OMDB_API_KEY is not set');
}
```

This makes configuration problems visible during development and testing.